src/minerva_lib/blend.py

```diff
@@ -70,33 +72,15 @@ def linear_bgr(channels):
     shape_color = shape + (3,)
 
     # Final buffer for blending
-    image_buffer = np.zeros(shape_color, dtype=np.float32)
+    out_buffer = np.zeros(shape_color, dtype=np.float32)
 
-    # Process all channels
+    # rescaled images and normalized colors
     for channel in channels:
 
-        image = channel['image']
-        color = channel['color']
-        min_ = channel['min']
-        max_ = channel['max']
-
-        # Scale the dynamic range
-        img_ranged = to_f32(image)
-
-        # Maximum color for this channel
-        avg_factor = 1.0 / num_channels
-        color_factor = color * avg_factor
-
-        # Fraction of full range
-        clip_size = max_ - min_
-
-        # Apply the range
-        img_ranged[(img_ranged < min_) | (img_ranged > max_)] = min_
-        img_norm = (img_ranged - min_) / clip_size
-
-        # Add the colored data to the image
-        y_shape, x_shape = img_norm.shape
-        img_color = f32_to_bgr(img_norm, color_factor)
-        image_buffer[0:y_shape, 0:x_shape] += img_color
+        # Add all three channels to output buffer
+        args = map(channel.get, ['image', 'color', 'min', 'max'])
+        out_buffer = composite_channel(out_buffer, *args, out=out_buffer)
```

Comment:
Just remove the `out_buffer =` assignment. Using the `out=` kwarg is understood to modify `out_buffer` in place.